kedro-datasets/kedro_datasets/spark/spark_dataset.py

```diff
@@ -290,15 +281,10 @@ def __init__(  # pylint: disable=too-many-arguments
         exists_function = None
         glob_function = None
 
-        if fs_prefix in ("s3a://", "s3n://"):
```

Comment:
This is also a sign of legacy, since `fs_prefix` can be an empty string. `"" in ("s3a://")` will also regard as `True` and thus it always uses `S3FileSystem`. I wonder why no one ever reports this bug. Maybe all our users are using Spark on s3 or `dbfs`?

In general I found too many legacy in this dataset that deserve overhaul at some point, the trickiest part is how do we make sure it works on all platform and test it properly.